src/main/java/ch/njol/skript/bukkitutil/block/NewBlockCompat.java

```diff
@@ -104,7 +116,26 @@ public MatchQuality match(BlockValues other) {
 				return MatchQuality.DIFFERENT;
 			}
 		}
-		
+
+		@Override
+		public Fields serialize() {
+			Fields fields = new Fields();
+			fields.putObject("data", data.getAsString());
+			fields.putPrimitive("isDefault", isDefault);
```

Comment:
Might be a good idea to assert / throw an exception in the constructor of NewBlockValues if `type` doesn't match the Material of `data`